### ISSUE TITLE
core-ajax-dart always sends a SYNCHRONOUS request by default, instead of ASYNC

### ISSUE THREAD

**Issue by DartBot** [dart-lang/polymer-dart#252]
_Originally opened as dart-lang/sdk#20246_

*This issue was originally filed by alexander.schaefe...&#064;gmail.com*
_____
**What steps will reproduce the problem?**

- Send any ajax request via core-ajax-dart using: 
&quot;ajax.go()&quot;

What is the expected output? 

- That the request sent is ASYNCHRONOUS

What do you see instead?

- The request sent is SYNCHRONOUS

SOLUTION/CAUSE OF THE PROBLEM:
in &quot;core_ajax_dart.dart&quot; Line 283, the 'sync' parameter is not passed:

&quot; return url == null ? null : this.xhr.request(
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;url: url,
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;method: method,
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;headers: headers,
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;body: body,
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;params: params,
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;withCredentials: withCredentials,
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;responseType: responseType,
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;callback: this.receive
&nbsp;&nbsp;&nbsp;&nbsp;);&quot;

Therefore, in &quot;core_xhr_dart.dart&quot; Line 64, the request will default to 'async = true', because 'sync == null':

&quot;var async = sync == false;&quot;

**Please include the following information:**
- Dart SDK version: 1.5.3
- polymer package version: any  (0.11.0+5)
- core_elements: 0.0.6
- Operating system: MacOS X 10.9.4
- Browser (if any): Dartium

**Comment by DartBot** [comment]
*This comment was originally written by alexander.schaefe...&#064;gmail.com*
_____
CORRECTION:

I meant to say that it incorrectly defaults to 'async = false':

\* Therefore, in "core_xhr_dart.dart" Line 64, the request will default to 'async = false'

**Comment by DartBot** [comment]
<a href="https://github.com/rafael-brandao"><img src="https://avatars.githubusercontent.com/u/2789070?v=3" align="left" width="48" height="48"hspace="10"></img></a> **Comment by [rafael-brandao](https://github.com/rafael-brandao)**

----

The simplest solution could be changing that line to:

"var async = (sync == null || sync == false);"

in "core_xhr_dart.dart".

**Comment by DartBot** [comment]
<a href="https://github.com/jakemac53"><img src="https://avatars.githubusercontent.com/u/984921?v=3" align="left" width="48" height="48"hspace="10"></img></a> **Comment by [jakemac53](https://github.com/jakemac53)**

----

moved to https://github.com/dart-lang/core-elements/issues/85
______
*Added [MovedToGithub](../labels/MovedToGithub) label.*
